otus-06/src/otus_06/homework.clj

```diff
@@ -25,36 +86,152 @@
 ;; 4|gum|1.25
 ;; 5|eggs|2.98
 ;; 6|jacket|42.99
+(defn get-products
+  ([] (get-products nil nil))
+  ([filter-k filter-v]
+   (base-loader "homework/prod.txt" product-schema filter-k filter-v)))
+
+(comment
+  (get-products))
+; [{"prodID" "1", "itemDescription" "shoes", "unitCost" "14.96"}
+;  {"prodID" "2", "itemDescription" "milk", "unitCost" "1.98"}
+;  {"prodID" "3", "itemDescription" "jam", "unitCost" "2.99"}
+;  {"prodID" "4", "itemDescription" "gum", "unitCost" "1.25"}
+;  {"prodID" "5", "itemDescription" "eggs", "unitCost" "2.98"}
+;  {"prodID" "6", "itemDescription" "jacket", "unitCost" "42.99"}]
 
 ;; sales.txt: это данные для основной таблицы продаж. Схема:
 ;; <salesID, custID, prodID, itemCount>.
+(def sales-schema [:salesID :custID :prodID :itemCount])
 ;;
 ;; Примером дискового файла sales.txt может быть:
 ;; 1|1|1|3
 ;; 2|2|2|3
 ;; 3|2|1|1
 ;; 4|3|3|4
 
-;; Например, первая запись (salesID 1) указывает, что Джон Смит (покупатель 1) купил 3 пары обуви (товар 1).
+(defn get-sales
+  ([] (get-sales nil nil))
+  ([filter-k filter-v]
+   (base-loader "homework/sales.txt" sales-schema filter-k filter-v)))
+
+(comment
+  (get-sales))
+; [{"salesID" "1", "custID" "1", "prodID" "1", "itemCount" "3"}
+;  {"salesID" "2", "custID" "2", "prodID" "2", "itemCount" "3"}
+;  {"salesID" "3", "custID" "2", "prodID" "1", "itemCount" "1"}
+;  {"salesID" "4", "custID" "3", "prodID" "3", "itemCount" "4"}]
+
+;; Например, первая запись (salesID 1) указывает, что Джон Смит (покупатель 1)
+;; купил 3 пары обуви (товар 1).
 
 ;; Задача:
-;; Предоставить следующее меню, позволяющее пользователю выполнять действия с данными:
+;; Предоставить следующее меню, позволяющее пользователю выполнять действия с
+;; данными:
 
 ;; *** Sales Menu ***
 ;; ------------------
+
 ;; 1. Display Customer Table
+(defn display-customers
+  []
+  (doseq [line (for [customer (get-customers)
+                     :let [{:keys [name address phoneNumber]} customer]]
+                 (format "Name: %s, Address: %s, Phone: %s"
+                         name
+                         address
+                         phoneNumber))]
+    (println line)))
+
 ;; 2. Display Product Table
+(defn display-products
+  []
+  (doseq [line (for [product (get-products)
+                     :let [{:keys [itemDescription unitCost]} product]]
+                 (format "Name: %s, Cost: $%s" itemDescription unitCost))]
+    (println line)))
+
 ;; 3. Display Sales Table
+(defn display-sales
+  []
+  (let [customers (into {}
+                        (for [customer (get-customers)
+                              :let [{:keys [custID name]} customer]]
+                          [custID name]))
```

Comment:
You do know `zipmap`, I saw it above :)
```
(let [cs (get-customers)] (zipmap (map :custID cs) (map :name cs)))
```
But yes, it's a matter of taste

Reply:
Nice variant thanks 👍🏼